syft/execution/protocol.py

```diff
@@ -139,7 +141,8 @@ def build(self):
         self.toggle_tracing(True)
         self.is_building = True
 
-        results = self.forward(self.roles)
+        sorted_roles = [self.roles[name] for name in sorted(self.roles.keys())]
```

Comment:
If I understood correctly, the users will have to provide args in alphabetical order of the corresponding role id?

Reply:
Yeah, that might not be ideal. I suppose we could just make them match with the list of role names provided above.

Reply:
Do you think we can give `**self.roles` and then the user have to name the args as he named the roles?
Still not ideal but a bit better maybe. What do you think?